libraries/pybamm-developer/01_ode.md

```diff
@@ -0,0 +1,341 @@
+
+$$
+\begin{align*}
+\frac{dx_n}{dt} &= -\frac{I(t)}{Q_n}, \\
+\frac{dx_p}{dt} &= \frac{I(t)}{Q_p}, \\
+V(t) &= U_p(x_p) - U_n(x_n) - I(t)R, \\
+x_n(0) &= x_{n0}, \\
+x_p(0) &= x_{p0}, \\
+\end{align*}
+$$
+
+where $x_n$ and $x_p$ are the dimensionless stochiometries of the negative and positive electrodes, $I(t)$ is the current, $Q_n$ and $Q_p$ are the capacities of the negative and positive electrodes, $U_p(x_p)$ and $U_n(x_n)$ are the open circuit potentials of the positive and negative electrodes, and $R$ is the resistance of the battery.
+
+## PyBaMM variables
+
+The reservoir model has a number of output variables, which are either the state
+variables $x_n$ and $x_p$ that are explicitly solved for, or derived variables
+such as the voltage $V(t)$. 
+
+In PyBaMM a state variable can be defined using the
+[`pybamm.Variable`](https://docs.pybamm.org/en/stable/source/api/expression_tree/variable.html#variable)
+class. For example, if you wanted to define a state variable with name "x", you
+would write
+
+```python
+x = pybamm.Variable("x")
+```
+
+::::challenge{id="ode-variables" title="Define the state variables for the reservoir model"}
+
+Define the state variables for the reservoir model, including the stochiometries
+$x_n$ and $x_p$.
+
+:::solution
+```python
+x_n = pybamm.Variable("Negative electrode stochiometry")
+x_p = pybamm.Variable("Positive electrode stochiometry")
+```
+:::
+::::
+
+We will leave the derived variables like $V(t)$ for now, and come back to them
+later once we have defined the expressions for the ODEs.
+
+## PyBaMM parameters
+
+The reservoir model has a number of parameters that need to be defined. In PyBaMM a parameter can be defined using the [`pybamm.Parameter`](https://docs.pybamm.org/en/stable/source/api/expression_tree/parameter.html#parameter) class. For example, if you wanted to define a parameter with name "a", you would write
+    
+```python
+a = pybamm.Parameter("a")
+```
+
+The name of the parameter is used to identify it in the model, the name of the Python variable you assign it to is arbitrary.
+
+You can also define a parameter that is defined as a function using the [`pybamm.FunctionParameter`](ihttps://docs.pybamm.org/en/stable/source/api/expression_tree/parameter.html#pybamm.FunctionParameter) class, which is useful for time varying parameters such as the current, or functions like the OCV function parameters $U_p(x_p)$ and $U_n(x_n)$, which are both functions of the stochiometries $x_p$ and $x_n$. For example, to define a parameter that is a function of time, you would write
+
+```python
+I = pybamm.FunctionParameter("I", {"Time [s]": pybamm.t})
```

Comment:
The syntax for `FunctionParameter` might not be clear. Should we say that "I" is the name of the parameters and the argument are given as a dictionary of name: symbol?